dora/src/object.rs

```diff
@@ -517,7 +519,9 @@ impl Str {
 
             handle
         } else {
-            Ref::null()
+            let mut handle = str_alloc_heap(vm, 0);
+            handle.length = std::usize::MAX;
+            return handle;
```

Comment:
Will these hacks break anything in the short term until the code is replaced?

Reply:
Pretty sure the GC is going to crash on such an object. You could force a GC after this with `forceCollect` to test. 

What would you ideally do in such cases? I think throwing an exception makes sense here. This should already work from native [code](https://github.com/dinfuehr/dora/blob/master/dora/src/stdlib.rs#L105-L113).

Reply:
Might make sense to open separate PRs for these API changes.

Reply:
Basically I want to have an `Option` here (see the corresponding hack in https://github.com/dinfuehr/dora/pull/97/files#diff-57504af7150824bb788bd6c48e08ca9e) but I'm unsure whether it is a good idea to assume a specific memory representation.

If getting a valid Dora Option from Rust code is too inconvenient at the moment, maybe it would be possible to make the GC aware of this hack to keep things going.

Reply:
We just can't add such a hack to the codebase. Especially since it doesn't block this work at all. There are perfectly fine alternatives: 1) throw an exception 2) return empty string 3) add support for Option for native code 4) rewrite function in Dora.

Reply:
You are right. We could return an empty string if it failed, and check whether the input was also empty: If it was, everything is fine, otherwise we encountered an error. Does that sound good to you?

Reply:
@dinfuehr I removed the ugly hack in https://github.com/dinfuehr/dora/pull/97/commits/ef2c530aa078d6ad4d720c887d741dffacad85be. Does it look good do you?

Reply:
I would've preferred the exception, since it makes the failure explicit but this is also okay.

Reply:
Yes, that's possible. That's why I linked the [throw_native](https://github.com/dinfuehr/dora/blob/master/dora/src/stdlib.rs#L116-L126) function above. This function doesn't do anything useful, it's only here for testing this functionality. But if you pass `true` as argument it is going to throw an exception from native code.

Reply:
(Sorry for not explicitly mentioning that)

Reply:
Ah, now I saw it! Thanks, I will revise the code accordingly!